src/main/java/io/debezium/connector/yugabytedb/transforms/YBExtractNewRecordState.java

```diff
@@ -41,6 +74,11 @@ public R apply(final R record) {
             updatedValueForValue = (Struct) val.getSecond();
         }
 
+        if (convertDeleteToTombstone && isDeleteRecord(record)) {
+            LOGGER.trace("Converting delete record to tombstone as per configuration");
+            return ret.newRecord(ret.topic(), ret.kafkaPartition(), updatedSchemaForKey, updatedValueForKey, null, null, record.timestamp());
```

Comment:
Don't we need to send the original delete record as well?

Reply:
No, we do not need to send the original record. This SMT is used where we need record flattening and the purpose is to convert all delete records to tombstone records so that the SMT can finally consume them.